src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -259,11 +257,12 @@ static void RegisterMessageHandlers(MessageHandlerRegistry handlerRegistry, List
 
             foreach (var t in types.Where(IsMessageHandler))
             {
-                container.ConfigureComponent(t, DependencyLifecycle.InstancePerUnitOfWork);
+                serviceCollection.AddWithInterfaces(t, ServiceLifetime.Scoped);
```

Comment:
Making this just AddScoped just broke the DI signature approvals since 
```
NServiceBus.IHandleMessages`1[[NServiceBus.AcceptanceTests.Core.DependencyInjection.When_endpoint_is_warmed_up+SomeMessage, NServiceBus.AcceptanceTests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=null]] - Scoped
``` 
is no longer registered.

All other tests still pass